h3/bq/sql/AS_H3.sql

```diff
@@ -0,0 +1,68 @@
+-----------------------------------------------------------------------
+--
+-- Copyright (C) 2021 CARTO
+--
+-----------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION `@@BQ_PROJECTID@@.@@BQ_DATASET_H3@@.LONGLAT_ASH3`(longitude FLOAT64, latitude FLOAT64, resolution INT64)
+    RETURNS INT64
+    DETERMINISTIC
+    LANGUAGE js
+    OPTIONS (library=["@@H3_BQ_LIBRARY@@"])
+AS
+"""
+    if (longitude === null || latitude === null || resolution === null) {
+        return null;
+    }
+    const index = h3.geoToH3(Number(latitude), Number(longitude), Number(resolution));
+    if (index) {
+        return '0x' + index;
+    }
+    return null;
+""";
+
+CREATE OR REPLACE FUNCTION `@@BQ_PROJECTID@@.@@BQ_DATASET_H3@@.ST_ASH3`(geog GEOGRAPHY, resolution INT64)
+    RETURNS INT64
+AS
+(
+    `@@BQ_PROJECTID@@.@@BQ_DATASET_H3@@.LONGLAT_ASH3`(SAFE.ST_X(geog), SAFE.ST_Y(geog), resolution)
+);
+
+CREATE OR REPLACE FUNCTION `@@BQ_PROJECTID@@.@@BQ_DATASET_H3@@.__ST_ASH3_POLYFILL`(geojson STRING, _resolution INT64)
+    RETURNS ARRAY<INT64>
+    DETERMINISTIC
+    LANGUAGE js
+    OPTIONS (library=["@@H3_BQ_LIBRARY@@"])
+AS
+"""
+    if (!geojson || _resolution == null) {
+        return null;
+    }
+
+    const resolution = Number(_resolution);
+    if (resolution < 0 || resolution > 15) {
```

Comment:
In this kind of cases what do we want, to return nothing or throwing an error. I was throwing some errors in quadkey for wrong zooms since I thought it would be easier to track for the user.

Now that I see this, I will also include some more checks for null arguments.

Reply:
I chose returning `NULL` because the underlying library was already doing that in most cases, and in SQL is a huge pain when you run a function over millions of records and it's stopped because a single failure. BigQuery has `SAFE.` but UDFs don't have that option. In any case it could be changed in the future to something different.